server/util.go

```diff
@@ -249,3 +250,21 @@ func InitHTTPClient(svr *Server) error {
 	}}
 	return nil
 }
+
+// ValidateLabel checks the legality of the label.
+func ValidateLabel(s string) error {
```

Comment:
The error messages here don't include the label itself. Which may not be a problem. However, the current callers are looping over labels and not attaching the label string either, so the client won't be told which label was found to be invalid.
Since labels seem to be used as arrays one solution would be useful to add a `ValidateLabels` function that takes an array of labels, calls ValidateLabel, and returns an error with the label string attached.